api/types/resetpasswordtokensecrets.go

```diff
@@ -39,6 +41,24 @@ type ResetPasswordTokenSecrets interface {
 	CheckAndSetDefaults() error
 }
 
+// NewResetPasswordTokenSecrets creates an instance of ResetPasswordTokenSecrets
+func NewResetPasswordTokenSecrets(tokenID string) (ResetPasswordTokenSecretsV3, error) {
```

Comment:
nit: why not return a pointer? Isn't this implementing the `ResetPasswordToken` interface?